### DISCUSSION TITLE
sh in lane cannot execute watchbuild

### DISCUSSION THREAD

**Discussion by Hwan-seok** [fastlane/fastlane#19224] [General]
I have sh command in my fastfile but the watchbuild is not executed only at the fastfile but exactly same command works properly on my own terminal.

How can I solve this?

I am using 
1. mac bigsur
2. iterm(also tried on terminal)
3. zsh
4. watchbuild 0.5

### What I did
```rb
default_platform(:ios)

platform :ios do
  lane :release do
    sh 'watchbuild -a com.example.app -u my@apple.id -k my_apple_team_id --slack_url https://hooks.slack.com/services/hook_url'
  end

end
```

### Result
```sh
Exit status of command 'watchbuild ...' was 1 instead of 0.
/Library/Ruby/Site/2.6.0/rubygems.rb:281:in `find_spec_for_exe': can't find gem watchbuild (>= 0.a) with executable watchbuild (Gem::GemNotFoundException)
	from /Library/Ruby/Site/2.6.0/rubygems.rb:300:in `activate_bin_path'
	from /usr/local/bin/watchbuild:23:in `<main>'

```

**Comment by lucgrabowski** [comment]
When you run `watchbuild` in terminal, gem can be found in a directory listed in `PATH` environment variable and it works fine.
When you run it in lane with `sh`, `PATH` environment variable isn't passed, and gem cannot be found.
I think that you could try to call `sh` this way:
```ruby
sh "PATH=#{ENV["PATH"]} watchbuild -a com.example.app -u my@apple.id -k my_apple_team_id --slack_url https://hooks.slack.com/services/hook_url"
```

**Reply by Hwan-seok** [reply]
@lucgrabowski 
Thanks for your answer!
I tried what you wrote above but the same result came out.

The `PATH` passed successfully but it does not work either...(same error)

I think the `gem` is found according to as follows
`/Library/Ruby/Site/2.6.0/rubygems.rb:281:in 'find_spec_for_exe': ...`

But maybe exactly `watchbuild` is not found in the gem.

**Reply by lucgrabowski** [reply]
Could you check `watchbuild` version in Gemfile and Gemfile.lock? I'm wondering why the error says `0.a` in `find_spec_for_exe': can't find gem watchbuild (>= 0.a)`, Should it be 0.5?

**Reply by Hwan-seok** [reply]
My watchbuild version is 0.5(I got this from `watchbuild --version`) and it is globally installed.
Where can I find the system global Gemfile(.lock)?

**Reply by lucgrabowski** [reply]
I think that you should take a look into your gemfile and confirm the version definition of `watchbuild` as you got:
`/Library/Ruby/Site/2.6.0/rubygems.rb:281:in `find_spec_for_exe': can't find gem watchbuild (>= 0.a)` This `0.a` sounds like a mistyped 0.5. There is no system global `Gemfile.lock`. 
If there isn't a typo in Gemfile, I would suggest installing a ruby version manager, e.g. `rbenv`.
